### ISSUE TITLE
dials.image_viewer: Brightness control does not screen non-numeric input

### ISSUE THREAD

**Issue by dagewa** [dials/dials#373]
I can type e.g. `really bright` into the input field, as a result get
```
Traceback (most recent call last):
  File "/home/david/sw/cctbx/modules/dials/util/image_viewer/spotfinder_frame.py", line 1640, in OnUpdateCM
    self.collect_values()
  File "/home/david/sw/cctbx/modules/dials/util/image_viewer/spotfinder_frame.py", line 1612, in collect_values
    self.settings.brightness = int(self.brightness_txt_ctrl.GetValue())
ValueError: invalid literal for int() with base 10: 'really bright'
```

**Comment by dagewa** [comment]
This now works correctly, but setting the brightness now takes somewhat longer. The problem seems to be that the control sets each character as they are entered, so if I enter '100', the image viewer will go through setting '1', '10' and '100'

**Comment by ndevenish** [comment]
You'd rather it only changed on e.g. hitting return?

**Comment by dagewa** [comment]
I think either return, or click elsewhere so loses focus

-- David

**Comment by phyy-nx** [comment]
Hi, I get an exception when using the scroll bar on the settings window now:

Traceback (most recent call last):
  File "/reg/neh/home4/brewster/projects/phenix-src/dials/util/image_viewer/spotfinder_frame.py", line 1661, in OnUpdateBrightness
    self.OnUpdateCM(event)
  File "/reg/neh/home4/brewster/projects/phenix-src/dials/util/image_viewer/spotfinder_frame.py", line 1645, in OnUpdateCM
    self.collect_values()
  File "/reg/neh/home4/brewster/projects/phenix-src/dials/util/image_viewer/spotfinder_frame.py", line 1615, in collect_values
    self.brightness_txt_ctrl.ChangeValue(self.settings.brightness)
  File "/reg/g/psdm/sw/releases/ana-current/arch/x86_64-rhel7-gcc48-opt/python/wx-2.8-gtk2-unicode/wx/_controls.py", line 1762, in ChangeValue
    return _controls_.TextCtrl_ChangeValue(*args, **kwargs)
TypeError: String or Unicode type required

**Comment by graeme-winter** [comment]
Folks should we be using wxPython 2 or 3? I see @phyy-nx  you are using 2.8; bootstrap uses 3

**Comment by graeme-winter** [comment]
Guess issue here is SLAC python installation at /reg/g/psdm/sw/releases/ana-current/arch/x86_64-rhel7-gcc48-opt/

**Comment by ndevenish** [comment]
It looks like this (second issue) is caused by this bug: http://trac.wxwidgets.org/ticket/13455, fixed in 2.9.

**Comment by ndevenish** [comment]
@dagewa your issue should be fixed now.

If the other SLAC/wx2.8 thing is still something we need to fix, can we open a separate issue so this one can be closed?

**Comment by dagewa** [comment]
Thanks Nick, confirmed fix works for me :+1:

Also, agree, the SLAC thing needs its own issue @phyy-nx, I am closing this one.

**Comment by phyy-nx** [comment]
Thanks all.  Fixed my part of this for LCLS/wxPython 2.8 using a try/except.
